airbyte-cdk/python/airbyte_cdk/sources/streams/core.py

```diff
@@ -544,4 +544,18 @@ def configured_json_schema(self) -> Optional[Dict[str, Any]]:
 
     @configured_json_schema.setter
     def configured_json_schema(self, json_schema: Dict[str, Any]) -> None:
-        self._configured_json_schema = json_schema
+        self._configured_json_schema = self._filter_schema_invalid_properties(json_schema)
+
+    def _filter_schema_invalid_properties(self, json_schema: Dict[str, Any | Dict[str, Any]]) -> Dict[str, Any]:
```

Comment:
let's add a warning when filtering out fields so users know why they're not present in the destination

Reply:
Sure, I'm adding the following, let me know if you have desire to rephrase:

[{
  "type": "LOG",
  "log": {
    "level": "WARN",
    "message": "For ads_insights_action_type the following properties in the configured catalog were not found in the the current stream schema and will be filtered: ['labels', 'location', 'age_targeting', 'gender_targeting', 'estimated_ad_recall_rate', 'estimated_ad_recallers_lower_bound', 'estimated_ad_recallers_upper_bound', 'estimated_ad_recall_rate_lower_bound', 'estimated_ad_recall_rate_upper_bound']"
  }
}](url)